### DISCUSSION TITLE
Random replacement of prompt parts even for a single image generation?

### DISCUSSION THREAD

**Discussion by Tillerz** [vladmandic/sdnext#565] [Q&A]
Hi, 

is it possible to have replacements like [1|2|3|4] does for batches, but for when rendering just one single image (no batch or grid)? I'd basically just need the UI to randomize the pick from that list instead of starting with the first element each time.

Prompt Matrix, Prompt from File and X/Y/Z Plot with Prompt S/R all go into that direction, but all require you to either make a lot of copies of the whole prompt and add your changing stuff to each of those OR they will generate a big batch of images.

I do not create batches. I create one image and then fiddle around with the settings. And it would be fabulous if I could just have stuff like random haircolor or random skin color etc for a single image, too. :)

**Comment by vladmandic** [comment]
perhaps one of prompt based extensions has that functionality.

**Comment by tkiikt** [comment]
@Tillerz Maybe try this: [sd-dynamic-prompts](https://github.com/adieyal/sd-dynamic-prompts)
